### DISCUSSION TITLE
I2C and SoftwareSerial not working in ATtiny85 if used together

### DISCUSSION THREAD

**Discussion by blopa1961** [SpenceKonde/ATTinyCore#854] [Support / Q & A]
Hello. I designed a telemetry compass for my FrSky radio control. The prototype works flawlessly in an Arduino Nano, but I wanted to reduce the size so I designed a PCB and ported it to an ATtiny85.
You can find the description of the problem here:
https://www.rcgroups.com/forums/showthread.php?4537315-Porting-the-S-PORT-compass-to-ATtiny85
Is SoftwareSerial supposed to work while I use I2C? Are they sharing a timer or INT?
I did a lot of tests, still no clue why this doesn't work.
The S.PORT protocol uses THE SAME pin (inverted) for both TX and RX, can this be a problem?

**Comment by hmeijdam** [comment]
I think you are including the S-port library (by Tandango)? I use that as well, but you will quickly run out of dynamic memory on the AT85 when you want to do more than just creating a sensor object.
How much dynamic memory does it use on your Nano?

**Reply by blopa1961** [reply]
Aha! I never thought of that! Let me check. Thank you!

**Reply by blopa1961** [reply]
@hmeijdam You were right!
I solved the problem by trimming the Magnetometer library.

In case you are interested, I published the (working) project with schematic and PCB here:
https://www.rcgroups.com/forums/showthread.php?4535205-S-PORT-Compass-for-FrSky-telemetry

I'm a seasoned Arduino programmer, but this is my first ATtiny85 project and I'm not used to such a small RAM.

Thank you for pointing out the RAM limitation, I was debugging in the wrong path.
